### DISCUSSION TITLE
working-with-tar-files-in-dotnet/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#433] [Announcements]
# working-with-tar-files-in-dotnet/

In this post I show how to create, extract, and list tar files using the built-in tar support introduced in .NET 7, and discuss some of the limitations

https://andrewlock.net/working-with-tar-files-in-dotnet/

**Comment by KoalaBear84** [comment]
I would want to use it, but I can't because of the lack of Stream's. I use `TarInputStream` from SharpZipLib instead. Also the same for ZIP, streaming support is very limited. Most of the APIs are file based, but I wanted to use streams without saving it to disk first.

What IS good and easy is GZipStream and BrotliStream, those work without issues, and.. are streams.

**Reply by latop2604** [reply]
You can use ZipArchive with not seekable stream like network stream. But you have to create a wrapper/decorator stream that store and expose the current stream position.

**Reply by KoalaBear84** [reply]
OK, thanks, so it is possible, but not 'fully' out of the box. It would be good if they supported this a little better.
